projects/graph/adv/world.py

```diff
@@ -5,24 +5,39 @@ class World:
     def __init__(self):
         self.startingRoom = None
         self.rooms = {}
+        self.occupied = set()
 
-    def getRandomDirection(self, room):
+    def getRandomDirection(self, room, coords):
         dirs = []
-        if room.n_to is None:
+        if room.n_to is None and self._checkCoordinates(coords, 'n'):
```

Comment:
This is good, consider though that the room.x_to part may not be super useful. There are circumstances where room.x_to can be undefined but there is a still an adjacent room in the x direction. Consider this layout:

[_]-[_]
 |   |
[_] [_]
 |     |
[_] [_]
 |     |
[_] [X]

I suspect the formatting may get messed up when I submit this comment, but basically the room that I marked as X has a room to the west but it's not "connected" so room.w_to would be None even though there is an adjacent room. That's why using coordinates to check is preferred.